### PR TITLE
Pass authenticated parameters to messages

### PR DESCRIPTION
**Pull request by fridex**
## Related Issues and Dependencies

Depends-On: https://github.com/thoth-station/messaging/pull/391
Depends-On: https://github.com/thoth-station/messaging/pull/390
Depends-On: https://github.com/thoth-station/common/pull/1146
Depends-On: https://github.com/thoth-station/common/pull/1147
Depends-On: https://github.com/thoth-station/thoth-application/pull/1132
Depends-On: https://github.com/thoth-station/thoth-application/pull/1133

Fixes: https://github.com/thoth-station/thoth-application/issues/945

## This introduces a breaking change

- [x] No

### CODE REVIEW

File: thoth/user_api/api_v1.py [hunk 1]
```diff
@@ -252,14 +263,18 @@ def post_provenance_python(
...
+    if token is not None:
+        if Configuration.API_TOKEN != token:
+            return {"error": "Bad token supplied"}, 401
+
+        authenticated = True
+        for k in _PROVENANCE_CHECK_PROTECTED_FIELDS:
             if parameters[k] is not None:
-                response = {"error": "Attempted to set protected field without proper permissions."}
-                return response, 401
+                return {"error": f"Parameter {k!r} requires token to be set to perform authenticated request"}, 401
```
Comment (pacospace) [inline thoth/user_api/api_v1.py:277]: sorry I'm confused by the conditions: 
if token is set and token is configuration.api_token, what is a case where we hit the last return (return {"error": f"Parameter {k!r} requires token to be set to perform authenticated request"}, 401)?

Comment (pacospace) [reply thoth/user_api/api_v1.py:278]: Now I understand :) Thanks @fridex!!

**Comment by fridex** [comment]
> Depends-On: [thoth-station/common#1146](https://github.com/thoth-station/common/pull/1146)
> Depends-On: [thoth-station/common#1147](https://github.com/thoth-station/common/pull/1147)

Part of [thoth-common==0.28.0 release](https://github.com/thoth-station/common/pull/1150).

thoth-messaging PRs will need to go in to fully support this.

/hold

**Review by harshad16 — COMMENTED** [review]
/lgtm

**Comment by fridex** [comment]
> Depends-On: [thoth-station/messaging#391](https://github.com/thoth-station/messaging/pull/391)
> Depends-On: [thoth-station/messaging#390](https://github.com/thoth-station/messaging/pull/390)

Part of thoth-messaging==0.13.0: https://github.com/thoth-station/messaging/pull/396

**Review by harshad16 — COMMENTED** [review]
/lgtm

**Review by harshad16 — APPROVED** [review]

**Pull request merged**